gyrokinetic/apps/gk_species_bflux.c

```diff
@@ -681,19 +696,19 @@ gk_species_bflux_init(struct gkyl_gyrokinetic_app *app, void *species,
     for (int m=0; m<bflux->num_calc_moms; m++) {
       gk_species_moment_init(app, gk_s, &bflux->moms_op[m], bflux->calc_mom_names[m], false);
 
-      need_m2perp = (bflux->calc_mom_names[m] == GKYL_F_MOMENT_M2PERP)
+      need_m2perp = need_m2perp || ( (bflux->calc_mom_names[m] == GKYL_F_MOMENT_M2PERP)
         || (bflux->calc_mom_names[m] == GKYL_F_MOMENT_M2)
         || (bflux->calc_mom_names[m] == GKYL_F_MOMENT_M0M1M2)
         || (bflux->calc_mom_names[m] == GKYL_F_MOMENT_M0M1M2PARM2PERP)
-        || (bflux->calc_mom_names[m] == GKYL_F_MOMENT_HAMILTONIAN);
+        || (bflux->calc_mom_names[m] == GKYL_F_MOMENT_HAMILTONIAN));
       bflux->is_hamiltonian_mom[m] = bflux->calc_mom_names[m] == GKYL_F_MOMENT_HAMILTONIAN;
       bflux->a_hamiltonian_mom = bflux->a_hamiltonian_mom || bflux->is_hamiltonian_mom[m];
     }
 
     if (need_m2perp) {
       // For moments that contain M2perp=2*mu*B/m we must fill the ghost cell of B. Use the option
       // in bc_basic that fills the ghost cell by evaluating the skin cell at the boundary.
-      long buff_sz = 1;
+      long buff_sz = 0;
```

Comment:
this can cause a problem. Some machines don't like it when you allocate size 0 memory. It used to be 0 (either here or in gk_species), and I changed it to 1 after seeing some valgrind errors